File: CS/InfoPanels.Module/DatabaseUpdate/Updater.cs
using System;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Updating;
using DevExpress.Xpo;
using DevExpress.Data.Filtering;
using DevExpress.Persistent.BaseImpl;
using DevExpress.ExpressApp.Security;

namespace InfoPanels.Module.DatabaseUpdate {
    public class Updater : ModuleUpdater {
        public Updater(IObjectSpace objectSpace, Version currentDBVersion) : base(objectSpace, currentDBVersion) {
        }
        public override void UpdateDatabaseAfterUpdateSchema() {
            base.UpdateDatabaseAfterUpdateSchema();
            Person personMary = ObjectSpace.FindObject<Person>(
            CriteriaOperator.Parse("FirstName == 'Mary' && LastName == 'Tellitson'"));
            if (personMary == null) {
                personMary = ObjectSpace.CreateObject<Person>();
                personMary.FirstName = "Mary";
                personMary.LastName = "Tellitson";
                personMary.Email = "tellitson@example.com";
                personMary.Birthday = new DateTime(1980, 11, 27);
            }
            Person personJohn = ObjectSpace.FindObject<Person>(
CriteriaOperator.Parse("FirstName == 'John' && LastName == 'Nilsen'"));
            if (personJohn == null) {
                personJohn = ObjectSpace.CreateObject<Person>();
                personJohn.FirstName = "John";
                personJohn.LastName = "Nilsen";
                personJohn.Email = "NilsenJ@example.com";
                personJohn.Birthday = new DateTime(1981, 11, 27);
            }
        }
    }
}

File: VB/InfoPanels.Module/DatabaseUpdate/Updater.vb
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Updating
Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.ExpressApp.Security

Namespace InfoPanels.Module.DatabaseUpdate
    Public Class Updater
        Inherits ModuleUpdater

        Public Sub New(ByVal objectSpace As IObjectSpace, ByVal currentDBVersion As Version)
            MyBase.New(objectSpace, currentDBVersion)
        End Sub
        Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
            MyBase.UpdateDatabaseAfterUpdateSchema()
            Dim personMary As Person = ObjectSpace.FindObject(Of Person)(CriteriaOperator.Parse("FirstName == 'Mary' && LastName == 'Tellitson'"))
            If personMary Is Nothing Then
                personMary = ObjectSpace.CreateObject(Of Person)()
                personMary.FirstName = "Mary"
                personMary.LastName = "Tellitson"
                personMary.Email = "tellitson@example.com"
                personMary.Birthday = New Date(1980, 11, 27)
            End If
            Dim personJohn As Person = ObjectSpace.FindObject(Of Person)(CriteriaOperator.Parse("FirstName == 'John' && LastName == 'Nilsen'"))
            If personJohn Is Nothing Then
                personJohn = ObjectSpace.CreateObject(Of Person)()
                personJohn.FirstName = "John"
                personJohn.LastName = "Nilsen"
                personJohn.Email = "NilsenJ@example.com"
                personJohn.Birthday = New Date(1981, 11, 27)
            End If
        End Sub
    End Class
End Namespace
